app/store/actions/healthcareAuthorities/getHealthcareAuthoritiesAction.ts

```diff
@@ -45,6 +47,20 @@ const getHealthcareAuthoritiesAction = (
         usesCustomUrl: !!customYamlUrl, // Hack to hijack this action for custom YAML configs. This is for testing only.
       }),
     );
+
+    if (autoSubscriptionLocation) {
+      const localHealthAuthority = healthcareAuthorities.find((ha) =>
+        isLocationWithinBounds(ha, autoSubscriptionLocation),
+      );
+
+      if (localHealthAuthority) {
+        dispatch(
+          healthcareAuthorityAutoSubscribeAction({
```

Comment:
Instead of using a new action, can you fire the existing toggle action? These actions duplicate in intent. 
```js
toggleSelectedHealthcareAuthorityAction({ authority: localHealthAuthority, override: true})
```

Reply:
The motivation for adding this specific action was to bypass the middleware that fires off the intersection algo.

I agree that this is an unnecessary bloating of redux interface and would like to change this, although it would be good to find a different way for firing off the intersection algo since it would attempt to download all of that HA data right on the app start.

Reply:
Per your advice, this was improved by adding a `meta` field to the action creator. Now we can have more control over firing off the intersection calculation